### ISSUE TITLE
Error during multiple ft-transfer txs. Get channel does not exist error

### ISSUE THREAD

**Issue by andynog** [informalsystems/ibc-rs#980]
## Crate

ibc-relayer-cli

## Summary of Bug

Trying several ft-transfer txs sometimes get a channel does not exist error.

## Version

commit: 2a56c9924bcf6393c8a8045ecf8fdb8cbe16e403 
branch: andy/telemetry

## Steps to Reproduce

Running a `start-multi` between `ibc-0` and `ibc-1` getting some errors when doing `ft-transfer`. Can't pinpoint the root error but just changing the timeout and number of transactions parameters seems to cause this error sometimes. Had to stop the relayer and start again and it execute pending transactions.

Start the relayer
```hermes -c ./config.toml start-multi```

Send a some ft-transfer tx
```
hermes -c ./config.toml tx raw ft-transfer ibc-1 ibc-0 transfer channel-0 9999 -o 1000 -n 3
hermes -c ./config.toml tx raw ft-transfer ibc-1 ibc-0 transfer channel-0 9999 -o 1 -n 10
```

Then sometimes get this error:

```
May 23 08:36:10.050 ERROR ibc_relayer::worker::uni_chan_path: channel error:  failed during a query to chain id ibc-0 with underlying error: Query error occurred (failed to query for port 'transfer' & channel 'channel-0'): channel does not exist:
May 23 08:36:16.112  INFO ibc_relayer::worker::client: running client worker (misbehaviour and refresh) for ibc-0 -> ibc-1:07-tendermint-0
May 23 08:36:17.314  INFO ibc_relayer::worker::client: running client worker (misbehaviour and refresh) for ibc-1 -> ibc-0:07-tendermint-0
May 23 08:36:33.401  INFO ibc_relayer::link: [ibc-0 -> ibc-1] generate messages from batch with 3 events
May 23 08:36:33.402 ERROR ibc_relayer::worker::uni_chan_path: channel error:  failed during a query to chain id ibc-0 with underlying error: Query error occurred (failed to query for port 'transfer' & channel 'channel-0'): channel does not exist:
```

But re-starting the relayer seems to fix the pending txs
```
May 23 08:36:59.526  INFO ibc_relayer::event::monitor: starting event monitor chain.id=ibc-0
May 23 08:36:59.528  INFO ibc_relayer::event::monitor: starting event monitor chain.id=ibc-1
May 23 08:36:59.532  INFO ibc_relayer::worker::client: running client worker & initial misbehaviour detection for ibc-1 -> ibc-0:07-tendermint-0
May 23 08:36:59.535  INFO ibc_relayer::worker::client: running client worker & initial misbehaviour detection for ibc-0 -> ibc-1:07-tendermint-0
May 23 08:36:59.739  INFO ibc_relayer::link: [ibc-0 -> ibc-1] clearing pending packets from events before height Height { revision: 0, height: 15341 }
May 23 08:36:59.739  INFO ibc_relayer::link: [ibc-0 -> ibc-1] clearing old packets
May 23 08:36:59.795  INFO ibc_relayer::link: [ibc-0 -> ibc-1] generate messages from batch with 13 events
May 23 08:36:59.826  INFO ibc_relayer::link: [ibc-0 -> ibc-1] scheduling op. data with 10 msg(s) for Source chain (height 1-15328)
May 23 08:36:59.826  INFO ibc_relayer::link: [ibc-0 -> ibc-1] scheduling op. data with 3 msg(s) for Destination chain (height 0-15341)
May 23 08:36:59.827  INFO ibc_relayer::link: [ibc-0 -> ibc-1] finished clearing pending packets
May 23 08:36:59.828  INFO ibc_relayer::link: [ibc-0 -> ibc-1] relay op. data to Destination, proofs height 0-15340, (delayed by: 2.064671ms) [try 1/10]
May 23 08:36:59.828  INFO ibc_relayer::link: [ibc-0 -> ibc-1] prepending Destination client update @ height 0-15341
May 23 08:36:59.838  INFO ibc_relayer::link: [ibc-0 -> ibc-1] assembled batch of 4 message(s)
May 23 08:37:00.342  INFO ibc_relayer::link: [ibc-0 -> ibc-1] result events:
	UpdateClientEv(h: 1-15328, cs_h: 07-tendermint-0(0-15341))
	WriteAcknowledgementEv(h:1-15328, seq:271, path:channel-0/transfer->channel-0/transfer, toh:1-16304, tos:Timestamp(NoTimestamp)))
	WriteAcknowledgementEv(h:1-15328, seq:272, path:channel-0/transfer->channel-0/transfer, toh:1-16304, tos:Timestamp(NoTimestamp)))
	WriteAcknowledgementEv(h:1-15328, seq:273, path:channel-0/transfer->channel-0/transfer, toh:1-16304, tos:Timestamp(NoTimestamp)))


May 23 08:37:00.342  INFO ibc_relayer::link: [ibc-0 -> ibc-1] success
May 23 08:37:00.342  INFO ibc_relayer::link: [ibc-0 -> ibc-1] relay op. data to Source, proofs height 1-15327, (delayed by: 515.955981ms) [try 1/10]
May 23 08:37:00.342  INFO ibc_relayer::link: [ibc-0 -> ibc-1] prepending Source client update @ height 1-15328
May 23 08:37:00.363  INFO ibc_relayer::link: [ibc-0 -> ibc-1] assembled batch of 11 message(s)
May 23 08:37:00.562  INFO ibc_relayer::link: [ibc-1 -> ibc-0] clearing pending packets from events before height Height { revision: 1, height: 15328 }
May 23 08:37:00.563  INFO ibc_relayer::link: [ibc-1 -> ibc-0] clearing old packets
May 23 08:37:00.683  INFO ibc_relayer::link: [ibc-0 -> ibc-1] result events:
	UpdateClientEv(h: 0-15342, cs_h: 07-tendermint-0(1-15328))
	TimeoutPacketEv(h:0-15342, seq:261, path:channel-0/transfer->channel-0/transfer, toh:1-15284, tos:Timestamp(NoTimestamp)))
	TimeoutPacketEv(h:0-15342, seq:262, path:channel-0/transfer->channel-0/transfer, toh:1-15284, tos:Timestamp(NoTimestamp)))
	TimeoutPacketEv(h:0-15342, seq:263, path:channel-0/transfer->channel-0/transfer, toh:1-15284, tos:Timestamp(NoTimestamp)))
	TimeoutPacketEv(h:0-15342, seq:264, path:channel-0/transfer->channel-0/transfer, toh:1-15284, tos:Timestamp(NoTimestamp)))
	TimeoutPacketEv(h:0-15342, seq:265, path:channel-0/transfer->channel-0/transfer, toh:1-15284, tos:Timestamp(NoTimestamp)))
	TimeoutPacketEv(h:0-15342, seq:266, path:channel-0/transfer->channel-0/transfer, toh:1-15284, tos:Timestamp(NoTimestamp)))
	TimeoutPacketEv(h:0-15342, seq:267, path:channel-0/transfer->channel-0/transfer, toh:1-15284, tos:Timestamp(NoTimestamp)))
	TimeoutPacketEv(h:0-15342, seq:268, path:channel-0/transfer->channel-0/transfer, toh:1-15284, tos:Timestamp(NoTimestamp)))
	TimeoutPacketEv(h:0-15342, seq:269, path:channel-0/transfer->channel-0/transfer, toh:1-15284, tos:Timestamp(NoTimestamp)))
	TimeoutPacketEv(h:0-15342, seq:270, path:channel-0/transfer->channel-0/transfer, toh:1-15284, tos:Timestamp(NoTimestamp)))


May 23 08:37:00.683  INFO ibc_relayer::link: [ibc-0 -> ibc-1] success
May 23 08:37:00.698  INFO ibc_relayer::link: [ibc-1 -> ibc-0] received from query_txs []
May 23 08:37:00.698  INFO ibc_relayer::link: [ibc-1 -> ibc-0] finished clearing pending packets
```
## Acceptance Criteria

____

## For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate milestone (priority) applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by ancazamfir** [comment]
Yes we started to see this after #923. It used to be that a channel query that returned no bytes was treated as a channel in "Uninitialized" state. After this PR it returns error.

The error you see happens during a flow like this:
- get latest chain height (this comes from the `status()` RPC)
- query a channel at that height (this comes from an ABCI query, i.e. app state query)

Sometimes the last step fails and I cannot explain why. I have seen this before and looks like a glitch in the consistency of the two pieces of information we query.

However hermes should retry and recover. You are not seeing this? I understand restart also fixes it but we want to recover via retries.

BTW, I changed this back under #924 (channel worker) as it is important to distinguish between a query error (e.g. RPC error) and a channel that does not exist. However this is going to cause problems in the packet handling where the reason we query the channel state is to make sure it is not closed....so we will fail to detect that if the channel is closed but the query says "Uninitialized"...what a mess!

We need to put more thought into this.
